geoportal/c2cgeoportal_geoportal/lib/filter_capabilities.py

```diff
@@ -50,13 +50,14 @@
 
 
 @CACHE_REGION.cache_on_arguments()
-def wms_structure(wms_url, host, request):
-    url = urlsplit(wms_url)
-    wms_url = add_url_params(wms_url, {"SERVICE": "WMS", "VERSION": "1.1.1", "REQUEST": "GetCapabilities"})
+def wms_structure(wms_url: str, host: str, request: pyramid.request.Request) -> Dict[str, List[str]]:
+    url = Url(wms_url)
+    url.add_query({"SERVICE": "WMS", "VERSION": "1.1.1", "REQUEST": "GetCapabilities"})
+    wms_url = str(wms_url)
 
     # Forward request to target (without Host Header)
     headers = dict()
-    if url.hostname == "localhost" and host is not None:  # pragma: no cover
+    if url.netloc == "localhost" and host is not None:  # pragma: no cover
```

Comment:
Be careful, If I correctly remember, netloc can contain the port number.
Here it may be `mapserver:8080` for example, while hostname would be `mapserver`.
Not sure it can break something here, but to be sure to not break anything, you may add a hostname property in class Url.

Reply:
Oups, thanks :-)